pkg/reconciliation/reconcile_racks.go

```diff
@@ -2243,6 +2246,52 @@ func (rc *ReconciliationContext) fixMissingPVC() (bool, error) {
 	return false, nil
 }
 
+func (rc *ReconciliationContext) SetFullQueryLogging() result.ReconcileResult {
+	datacenter := rc.GetDatacenter()
+	var DCConfig map[string]interface{}
+	if err := json.Unmarshal(datacenter.Spec.Config, &DCConfig); err != nil {
+		rc.ReqLogger.Error(err, "error unmarshalling DC config JSON")
+		return result.RequeueSoon(2)
+	}
+	shouldFQLBeEnabled := false
+	if err := DCConfig["full_query_logging_options"]; err == nil {
+		serverMajorVersion, err := strconv.ParseInt(strings.Split(datacenter.Spec.ServerVersion, ".")[0], 10, 8)
+		if err != nil {
+			rc.ReqLogger.Error(err, "error parsing server major version. Can't enable full query logging without knowing this")
+			return result.RequeueSoon(2)
+		}
+		if serverMajorVersion < 4 {
+			err := goerrors.New("full query logging only supported on OSS Cassandra 4x+")
```

Comment:
Since fmt is already imported, one can use fmt.Errorf() here to avoid additional import with rename.